src/node_url_pattern.cc

```diff
@@ -553,7 +511,9 @@ void URLPattern::Exec(const FunctionCallbackInfo<Value>& args) {
     input_base = input_value.ToString();
     input = std::string_view(input_base);
   } else if (args[0]->IsObject()) {
-    input = URLPatternInit::FromJsObject(env, args[0].As<Object>());
+    auto maybeInput = URLPatternInit::FromJsObject(env, args[0].As<Object>());
+    if (!maybeInput.has_value()) return;
```

Comment:
It is really confusing why this function doesn't throw before returning whereas 5 lines below, we do throw an error. 

```suggestion
    if (!maybeInput) return;
```

Reply:
We return std::nullopt in cases where the underlying v8 calls failed and an error should have been thrown already.

Reply:
Can you add a comment to here as well?